pkg/v1/tkg/client/management_components.go

```diff
@@ -52,22 +61,126 @@ func (c *TkgClient) InstallManagementComponents(kubeconfig, kubecontext string)
 }
 
 func (c *TkgClient) getTKGPackageConfigValuesFile() (string, error) {
+	userProviderConfigValues, err := c.getUserConfigVariableValueMap()
+	if err != nil {
+		return "", err
+	}
+
+	valuesFile, err := managementcomponents.GetTKGPackageConfigValuesFileFromUserConfig(userProviderConfigValues)
+	if err != nil {
+		return "", err
+	}
+
+	return valuesFile, nil
+}
+
+func (c *TkgClient) getUserConfigVariableValueMap() (map[string]string, error) {
 	path, err := c.tkgConfigPathsClient.GetConfigDefaultsFilePath()
+	if err != nil {
+		return nil, err
+	}
+
+	return c.GetUserConfigVariableValueMap(path, c.TKGConfigReaderWriter())
+}
+
+func (c *TkgClient) getUserConfigVariableValueMapFile() (string, error) {
+	userConfigValues, err := c.getUserConfigVariableValueMap()
 	if err != nil {
 		return "", err
 	}
 
-	userProviderConfigValues, err := c.GetUserConfigVariableValueMap(path, c.TKGConfigReaderWriter())
+	configBytes, err := yaml.Marshal(userConfigValues)
 	if err != nil {
 		return "", err
 	}
 
-	valuesFile, err := managementcomponents.GetTKGPackageConfigValuesFileFromUserConfig(userProviderConfigValues)
+	prefix := []byte(`#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+`)
+	configBytes = append(prefix, configBytes...)
+
+	configFile, err := utils.CreateTempFile("", "*.yaml")
+	if err != nil {
+		return "", err
+	}
+	err = utils.WriteToFile(configFile, configBytes)
+	if err != nil {
+		return "", err
+	}
+	return configFile, nil
+}
+
+func (c *TkgClient) getKappControllerConfigFile() (string, error) {
+	kappControllerPackageImage, err := c.tkgBomClient.GetKappControllerPackageImage()
 	if err != nil {
 		return "", err
 	}
 
-	return valuesFile, nil
+	path, err := c.tkgConfigPathsClient.GetTKGProvidersDirectory()
+	if err != nil {
+		return "", err
+	}
+	kappControllerValuesDirPath := filepath.Join(path, "kapp-controller-values")
+
+	userConfigValuesFile, err := c.getUserConfigVariableValueMapFile()
+	if err != nil {
+		return "", err
+	}
+
+	log.Infof("userConfigValuesFile: %v", userConfigValuesFile)
+
+	return ProcessKappControllerPackage(kappControllerPackageImage, userConfigValuesFile, kappControllerValuesDirPath)
+}
+
+func ProcessKappControllerPackage(kappControllerPackageImage, userConfigValuesFile, kappControllerValuesDirPath string) (string, error) {
+	kappControllerValuesFile, err := GetKappControllerConfigValuesFile(userConfigValuesFile, kappControllerValuesDirPath)
+	if err != nil {
+		return "", err
+	}
+
+	log.V(3).Infof("kappControllerValuesFile: %v", kappControllerValuesFile)
+
+	configBytes, err := carvelhelpers.ProcessCarvelPackage(kappControllerPackageImage, kappControllerValuesFile)
+	if err != nil {
+		return "", err
+	}
+
+	configFile, err := utils.CreateTempFile("", "")
```

Comment:
nit: should we make an attempt to clean up the temp files created?

Reply:
updated to delete temp file on success. But will be keeping the temp file on failures